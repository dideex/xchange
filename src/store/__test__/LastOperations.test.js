import LO from '../LastOperations'
import mockio, {serverSocket, cleanUp} from 'socket.io-client'

describe('Last operations tests', () => {
  let store
  beforeEach(() => {
    store = new LO()
  })

  it('init store', () => {
    expect(store.loading).toBeTruthy()
    expect(store.data.length).toBe(0)
  })

  it('Initial socket data', () => {
    expect(store.loading).toBeTruthy()
    serverSocket.emit('message', {type: 'init', data: [1, 2, 3, 4]})
    expect(store.data).toMatchSnapshot()
    expect(store.loading).toBeFalsy()
    serverSocket.emit('message', {type: 'init', data: [4, 3, 2, 1]})
    expect(store.data).toMatchSnapshot()
    expect(store.loading).toBeFalsy()
  })

  it('Braodcast new oparation', () => {
    serverSocket.emit('message', {type: 'broadcast', order: {data: 'new operation 1'}})
    expect(store.data).toMatchSnapshot()
    serverSocket.emit('message', {type: 'broadcast', order: {data: 'new operation 2'}})
    expect(store.data).toMatchSnapshot()
    serverSocket.emit('message', {type: 'broadcast', order: {data: 'new operation 3'}})
    expect(store.data).toMatchSnapshot()
    expect(store.loading).toBeFalsy()
  })
})
