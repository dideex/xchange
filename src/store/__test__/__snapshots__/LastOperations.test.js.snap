// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last operations tests Braodcast new oparation 1`] = `
Array [
  Object {
    "data": "new operation 1",
  },
]
`;

exports[`Last operations tests Braodcast new oparation 2`] = `
Array [
  Object {
    "data": "new operation 1",
  },
  Object {
    "data": "new operation 2",
  },
]
`;

exports[`Last operations tests Braodcast new oparation 3`] = `
Array [
  Object {
    "data": "new operation 1",
  },
  Object {
    "data": "new operation 2",
  },
  Object {
    "data": "new operation 3",
  },
]
`;

exports[`Last operations tests Initial socket data 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`Last operations tests Initial socket data 2`] = `
Array [
  4,
  3,
  2,
  1,
]
`;
